### DISCUSSION TITLE
Update metadata when it's a list to add another element and not replace them

### DISCUSSION THREAD

**Discussion by MrYod** [mProjectsCode/obsidian-meta-bind-plugin#215] [Q&A] [answered]
I have a metadata property, named "places" (translated, but irrelevant), that is a list of internal links to other notes that represent those places.
Now: I wanted to make a button that creates another place note and add that link in the frontmatter, but the usual updateMetadata replaecs the specific property with the one specified in "value" while, with "evaluate: true", the list breaks.

Example:
with
```meta-bind-button
label: Add place
hidden: false
class: ""
tooltip: ""
id: ""
style: default
actions:
  - type: updateMetadata
    bindTarget: places
    evaluate: true
    value: String(x) + "\n - '[[2]]'"
```
or value: x + ["\n - [[2]]"]
or value: String(x) + ["\n - [[2]]"]
or value: x + "\n - '[[2]]'"
or any other iteration of "", '' and [], for that matter

and no "places" in the frontmatter the property becomes:
```YAML
places: |-
  undefined
   - '[[2]]'
 ```

with no elements in "places" becomes:
 ```YAML
places: |-
  null
   - '[[2]]'
 ```

and with places: "[[1]]" the property becomes:
 ```YAML
places: |-
  [[1]]
   - [[2]]
```

without the "\n" there won't be a list and everything'll be a one-liner.
And with [] around the variable (like: value: [x] + ["\n - [[2]]"]) the button doesn't even appear.

So: how do I actually **update** a list property in the frontmatter?

**Comment by mProjectsCode** [comment, marked as answer]
try something like `value: "[...x, 'new value']"`

**Reply by MrYod** [reply]
"error while running action TypeError: x is not iterable"
with `value: "[...x, '[[2]]']"` when "places" doesn't exist or is empty but it works when there's already an element.

So: partial solution, thanks. 👍

Now: is it possible to solve that even when the list is empty or I'll have to find a workaroud (like putting an empty link in "places" in the template)?

**Reply by MrYod** [reply]
No, wait: with `places: []` it works properly.

So I'd say complete solution, keeping that in mind.

**Reply by MrYod** [reply]
Ok, not quite complete solution, after all: if I put `places: []` in the template, then Obsidian removes the "[]" and, in the new note I get only `places:`

Resulting in an "error while running action TypeError: x is not iterable" when I press the Add Place button.

So I have to put `places: "[]"` in the template with the result that, after clicking on the Add Place button, I'll get:
```YAML
places: 
 - "[]"
 - "[[NewPlaceLink|TemporaryAlias]]"
```
so I'll have to manually delete the first entry. 🤔

EDIT: Obsidian deletes the "[]" at the first change in any frontmatter field from "File properties" window.

**Reply by mProjectsCode** [reply]
Something like this should work 
```yaml
label: Add Current Time to List
hidden: false
class: ""
tooltip: ""
id: ""
style: primary
actions:
  - type: updateMetadata
    bindTarget: someList
    evaluate: true
    value: "x == null ? [Date.now()] : [...x, Date.now()]"

```

**Reply by MrYod** [reply]
Yes: it does.

Thanks.
